js/pluginDefinitionValidation.js

```diff
@@ -0,0 +1,127 @@
+
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+'use strict';
+
+const zluxUtil = require('./util');
+const bootstrapLogger = zluxUtil.loggers.bootstrapLogger;
+const fs = require('fs');
+
+// **************************************************************************************************************************************************
+// Set up JSON Schema using ajv library
+
+const Ajv = require('ajv')
+const ajv = new Ajv({ allErrors: true, jsonPointers: true, verbose: true, $data: true })
+require('ajv-errors')(ajv)
+
+const Schemas = require('../lib/schema/Schemas');
+const pluginDefinitionSchema = Schemas.pluginDefinitionSchema;
+
+// **************************************************************************************************************************************************
+// Validates the plugin definition based on the json schema defined above
+// Uses custom error messages in conjunction with the error messages defined in the schema
+
+module.exports.validatePluginDef = function validatePluginDef(pluginDef) {
+    const validate = ajv.compile(pluginDefinitionSchema)
```

Comment:
You should not be compiling per plugin def. Compile each schema that exists once.
See example from https://www.npmjs.com/package/ajv
```
var validate = ajv.compile(schema);
console.log(validate(2.01)); // true
console.log(validate(3.99)); // true
console.log(validate(2)); // false
console.log(validate(4)); // false
```